hibernation-setup-tool.c

```diff
@@ -1535,13 +1539,15 @@ static int handle_pre_systemd_suspend_notification(const char *action)
                     log_fatal("Couldn't remove /tmp/hibernation-setup-tool directory before proceeding");
                 }
 
-                log_info("/tmp/hibernation-setup-tool exists and isn't a directory! Removing it and trying again (try %d)", try);
+                else {
+                    log_info("/tmp/hibernation-setup-tool exists and isn't a directory! Removing it and trying again (try %d)", try);
 
-                if (!unlink("/tmp/hibernation-setup-tool"))
-                    continue;
+                    if (!unlink("/tmp/hibernation-setup-tool"))
+                        continue;
 
-                notify_vm_host(HOST_VM_NOTIFY_PRE_HIBERNATION_FAILED);
-                log_fatal("Couldn't remove the file: %s, giving up", strerror(errno));
+                    notify_vm_host(HOST_VM_NOTIFY_PRE_HIBERNATION_FAILED);
+                    log_fatal("Couldn't remove the file: %s, giving up", strerror(errno));
+                }
             }
 
             log_info("/tmp/hibernation-setup-tool couldn't be found but mkdir() told us it exists, trying again (try %d)", try);
```

Comment:
I can understand rerunning this code once, but not ten times? There must be something I'm missing I am going to leave this code for now.